### PR TITLE
Switch to using the Fuchsia buildtools.

### PR DESCRIPTION
**Pull request by chinmaygarde**
* Uses the Clang from Fuchsia buildtools and gets rid of ours from Chromium. Clang is now 4.0 instead of 3.7.
* Will pull libcxx, libcxxabi and unwind from Fuchsia and use that on Linux hosts before flipping the switch back on on C++14.
* The Clang plugins have been disabled for now. If we really need them, I will update them as well and make sure they work with the newer Clang.
* Clang finds more warnings in our codebase. Have disabled some in Harbuzz. Fixes in our codebase (in a separate patch in the engine.)

Have only switched the Darwin targets. Just for initial review. Will patch all platforms before landing.

### CODE REVIEW

**Comment by abarth** [comment]
rslgtm

**Comment by chinmaygarde** [comment]
Added a patch that updates the Linux build to use the libcxx bundled with the Fuchsia buildtools. Also re-enabled C++14.

cc @jamesr

**Comment by jamesr** [comment]
What's the GitHub voodoo to clone this locally?

File: build/config/compiler/BUILD.gn [hunk 1]
```diff
@@ -668,6 +618,20 @@ config("runtime_library") {
...
+
+  # Linux standard liburary setup.
+  if (is_linux) {
+    cflags_cc += [
+      "-stdlib=libc++"
+    ]
+    ldflags += [
+      "-stdlib=libc++",
+      "-static-libstdc++",
+      "-Wl,-Bstatic", "-lc++abi",
```
Comment (jamesr) [inline build/config/compiler/BUILD.gn]: I expect we'll be able to remove some or all of these lines soon, assuming Petr cleans up our toolchain distro. Can you leave a TODO or some other comment here reminding the next buildtools roller to reconsider these?

File: build/config/compiler/BUILD.gn [hunk 2]
```diff
@@ -668,6 +618,20 @@ config("runtime_library") {
...
+      "-Wl,-Bdynamic",
```
Comment (jamesr) [inline build/config/compiler/BUILD.gn]: nit: please put this up on the previous line, as it's really resetting state from that.

Comment (chinmaygarde) [inline build/config/compiler/BUILD.gn]: Done.

File: build/config/compiler/BUILD.gn (hunk 1 above)
Comment (chinmaygarde) [inline build/config/compiler/BUILD.gn]: Ok. Will file a ticket to track.

**Comment by jamesr** [comment]
b4eb7a12adbc194b4f715a07784d9b48e7c45db0 LGTM. pretty sure the rest is fine, but github's UI won't show it to me :P

**Comment by chinmaygarde** [comment]
Cool, I just need to test the Android x86 and x64 builds. Then a deps update.

**Pull request merged**